vumi/persist/tests/test_model.py

```diff
@@ -560,6 +561,25 @@ def test_index_keys(self):
         self.assertEqual(sorted(keys), ["foo1", "foo2"])
 
         keys = yield indexed_model.index_keys('b', None)
+        self.assertEqual(keys, [])
+
+    @Manager.calls_manager
+    def test_index_keys_store_none_for_empty(self):
+        self.patch(fields, "STORE_NONE_FOR_EMPTY_INDEX", True)
+        indexed_model = self.manager.proxy(IndexedModel)
+        yield indexed_model("foo1", a=1, b=u"one").save()
+        yield indexed_model("foo2", a=2, b=u"one").save()
+        yield indexed_model("foo3", a=2, b=None).save()
+
+        keys = yield indexed_model.index_keys('a', 1)
+        self.assertEqual(keys, ["foo1"])
+        # We should get a list object, not an IndexPage wrapper.
+        self.assertTrue(isinstance(keys, list))
```

Comment:
These tests are unmodified copies of what we had before except for the monkey patch at the top. I think it's still worth testing all of the original cases while we have the hack in the codebase, because the tests shouldn't assume that setting `STORE_NONE_FOR_EMPTY_INDEX = True` won't affect non-`None` values.